### ISSUE TITLE
Bad Apache settings in apache container block .htaccess

### ISSUE THREAD

**Issue by bondjimbond** [Islandora-Collaboration-Group/ISLE#362]
#### Issue description
In the Apache container, `/etc/apache2/apache2.conf` is using `AllowOverride None` and blocking `.htaccess` from being properly used. This is in a Staging implementation.

#### For an issue, describe steps to reproduce the issue

System setup (OS information, software versions, etc): RedHat, fresh ISLE install.

Steps:
1. Follow all steps for a staging migration install
2. Note that Drupal paths don't work when testing Drupal
3. Check `/etc/apache2/apache2.conf` and see the problem:

```
<Directory /var/www/>
	Options Indexes FollowSymLinks
	AllowOverride None
	Require all granted
</Directory>
```

##### What's the expected result?
Should be `AccessOverride All`

##### What's the actual result?
`AccessOverride None`

#### Additional details / screenshots
Because of this, the website simply doesn't work. If you change the setting to `AccessOverride All` and restart Apache, suddenly the website is fine.

But none of these settings were configured on installation; so far as I can tell they're coming from the container images. This needs to be fixed.

**Comment by g7morris** [comment]
Hi @bondjimbond 

I'm not sure where you are getting that Apache vhost setting This is setting the apache image uses by default https://github.com/Islandora-Collaboration-Group/isle-apache/blob/master/rootfs/etc/confd/templates/apache/site_template.conf.tpl#L10

This isn't something that an ISLE user sets by default. I suspect something from your two Apache setup is coming back into play here. 

I'd recommend that you build an ISLE host server with no other services only Docker
and then deploy the ISLE images and see how you fare.

Typically having an additional non-Docker Apache or MySQL server installed on the ISLE host server in addition to the ISLE containers causes all sorts of fights and this type of behavior

**Comment by bondjimbond** [comment]
As @noahwsmith found: https://github.com/Islandora-Collaboration-Group/isle-apache/blob/master/rootfs/etc/confd/templates/apache/apache2.conf.tpl#L172

Turns out there **is** an override putting the wrong settings in.

Side note: why are there two templates?

**Comment by noahwsmith** [comment]
Looking at one of our prod systems, I can see that the line in question is present:
`<Directory /var/www/>
	Options Indexes FollowSymLinks
	AllowOverride None
	Require all granted
</Directory>`

so why don't we see the same problem you do? I think it's because the main vhost (/etc/apache2/sites-enabled/islandora-site.conf) overrides this:

`	<Directory /var/www/html/>
		Require all granted
		Options FollowSymlinks
		AllowOverride all
	</Directory>`

So that makes me think that your vhost isn't getting picked up.

**Comment by noahwsmith** [comment]
That said, I don't see HARM in accepting the PR. 

But I think it patches the symptom, not the root cause (which we don't yet understand)

**Comment by noahwsmith** [comment]
thanks for helping debug on Slack, @br2490

**Comment by noahwsmith** [comment]
Ben suggested:

you can bind mount the template file so when confd runs (the processor of the template) it’s your version that’s used. So copy down https://github.com/Islandora-Collaboration-Group/isle-apache/blob/master/rootfs/etc/confd/templates/apache/site_template.conf.tpl and bind mount your version to `etc/confd/templates/apache/site_template.conf.tpl`

**Comment by bondjimbond** [comment]
@noahwsmith @br2490 Tried the bind mount trick -- it did not work.

**Comment by bondjimbond** [comment]
And it is definitely using my mounted copy of the template; I added a test comment in there and it's present if I bash into the apache machine.

Given that the PR does no harm, it would be very helpful to cut off that particular source of problems.

**Comment by g7morris** [comment]
@bondjimbond  As I've commented on your [PR](https://github.com/Islandora-Collaboration-Group/isle-apache/pulls I've started the ISLE 1.4.1 release build process and will include this PR for testing. I'll update this ticket in the next week with results. Our plan is to release ISLE 1.4.1 for the end of January but next week during the testing process, I'll loop you in on using a test image to try out for the fix.

**Comment by bondjimbond** [comment]
Thanks!

**Comment by g7morris** [comment]
Hi @bondjimbond 

I've completed the ISLE 1.4.1 build process, mnerged in your PR to the test code and have an Apache image for you to test if you're up for it. 

I recommend that:

a) shutdown the containers on the running system you want to test on. `docker-compose down`

b) You edit the appropriate / involved `docker-compose.*.yml` file with the following in the apache service section:

```bash
apache:
    image: islandoracollabgroup/isle-mysql:1.4.0

should become:

  apache:
    image: borndigital/isle-mysql:1.4.1-dev
```

c) `docker-compose pull` (pulls down new test image)

d) Fire up the containers: `docker-compose up -d`

e) QC / QA the site and report back on this issue / ticket if it is or isn't working. Thanks!

**Comment by bondjimbond** [comment]
Some confusion: 

```apache:
    image: islandoracollabgroup/isle-mysql:1.4.0

should become:

  apache:
    image: borndigital/isle-mysql:1.4.1-dev
```
Am I editing `apache: image: islandoracollabgroup/isle-apache:1.4.0` or mysql: image: islandoracollabgroup/isle-mysql:1.4.0`?

There is no mysql image under `apache:`, but if I try changing it to `isle-apache:1.4.1-dev` I get the error when I run `docker-compose pull`:

`ERROR: manifest for islandoracollabgroup/isle-apache:1.4.1-dev not found: manifest unknown: manifest unknown`

**Comment by g7morris** [comment]
@bondjimbond My apologies, I didn't notice the wrong image name in there. Lack of coffee. I've updated my comment above to the correct method. 

```bash
apache:
    image: islandoracollabgroup/isle-apache:1.4.0

should now become:

  apache:
    image: borndigital/isle-apache:1.4.1-dev
```

**Comment by bondjimbond** [comment]
Ah, now we're getting somewhere. Thanks, pull is in progress. Will report back with success/failure.

**Comment by bondjimbond** [comment]
Success!

- Modified `docker-compose.staging.yml` as instructed
- Removed bind mount to my custom template to use the one from the image
- Result: it works. https://sandbox.arcabc.ca/about

All good!

**Comment by g7morris** [comment]
Thanks for testing and confirming this @bondjimbond I'll be sure to merge in your PR to the official release. In addition to the fixes, thanks and appreciation of all your hard work, I've immortalized you as a contributor in the upcoming release notes. https://github.com/Born-Digital-US/ISLE/blob/ISLE-v.1.4.1-dev/docs/release-notes/release-1-4-1.md :)

**Comment by bondjimbond** [comment]
Aw, thanks. :)

**Comment by g7morris** [comment]
Closing ticket as @bondjimbond 's [commit](https://github.com/Islandora-Collaboration-Group/isle-apache/commit/7f94b6b6ee40fc39bd86addf312ae84b14072247) fixes this issue. Merged with 1.4.1 release. Thanks again @bondjimbond
